app/forms/draft_work_form.rb

```diff
@@ -175,4 +176,18 @@ def model_name
   def will_assign_doi?
     (collection.doi_option == 'depositor-selects' && assign_doi) || collection.doi_option == 'yes'
   end
+
+  # de-dupe keywords, preferring to throwing away the free text entry duplicate(s) without a URI
+  def dedupe_keywords
+    return if work_version.keywords.size.zero?
+
+    grouped_keywords = Keyword.where(work_version:).group_by(&:label) # group keywords by the label
+    grouped_keywords.each do |_label, group|
```

Comment:
Slight suggestion: `grouped_keywords.values.each`

Reply:
sounds good, in fact rubocop suggested `grouped_keywords.each_value` which I didn't know existed, and will use.